jetty-jmx/src/main/java/org/eclipse/jetty/jmx/ObjectMBean.java

```diff
@@ -573,41 +385,20 @@ public Object invoke(String name, Object[] params, String[] signature) throws MB
         }
     }
 
-    private static List<Class<?>> findInfluences(List<Class<?>> influences, Class<?> aClass)
-    {
-        if (aClass != null)
-        {
-            if (!influences.contains(aClass))
-            {
-                // This class is a new influence
-                influences.add(aClass);
-            }
-
-            // So are the super classes
-            influences = findInfluences(influences, aClass.getSuperclass());
-
-            // So are the interfaces
-            Class<?>[] ifs = aClass.getInterfaces();
-            for (int i = 0; ifs != null && i < ifs.length; i++)
-                influences = findInfluences(influences, ifs[i]);
-        }
-        return influences;
-    }
-
     /**
      * <p>Defines an attribute for the managed object using the annotation attributes.</p>
      *
-     * @param method the method on the managed objec
      * @param attributeAnnotation the annotation with the attribute metadata
+     * @param getter the method on the managed objec
      * @return an MBeanAttributeInfo with the attribute metadata
      */
-    private MBeanAttributeInfo defineAttribute(Method method, ManagedAttribute attributeAnnotation)
+    MBeanAttributeInfo defineAttribute(ManagedAttribute attributeAnnotation, Method getter)
```

Comment:
I think this method and `defineOperation()` should be moved to `MBeanContainer`, since you have moved the metadata discovery and building there.

Furthermore, we now have 2 caches: `MBeanContainer._metaData._getters` and here `this._getters`, along with the same for setters, etc.

Reply:
Yes.  I was trying to minimise changes so avoided looking into these methods, but agree they should move.